### DISCUSSION TITLE
Getting the "Adafruit ic2 16x2 RGB LCD PI Plate" working with the MiniDexed

### DISCUSSION THREAD

**Discussion by drhitchcockco** [probonopd/MiniDexed#349] [Q&A] [answered]
Hi there,

Freaking amazing project! I was up and running in minutes. Thanks for all the hard work you have all put into this!

Does anyone know how I might get the [Adafruit ic2 16x2 RGB LCD PI Plate](https://www.adafruit.com/product/1109) working with this? 

Thanks,
Benet.

**Comment by diyelectromusic** [comment, marked as answer]
From the schematic, it looks like everything on the display and IO is driven over I2C.  So the buttons won't work unless we are able to build in I2C support for the MCP23017 IO expander, and then it would need the specifics for this board to be configured.  You might get somewhere with the I2C display, as it looks like a HD44780 display which is supported, but it also looks like the backlight is controlled via GPIO pins from the MCP23017 too, so you wouldn't be able to turn on the backlight via software.

It might be possible to hardware mod the backlight display to fix it permanently on...

We don't really support displays with extra I2C control requirements, as there are so many variants out there!

So, if you wanted to build your own firmware to use it, you'd first have to work out how to drive the MCP23017 over I2C from the circle environment and then hook that into the uibuttons interface and hardwire in some code to turn on the backlight over I2C.  It is doable, but would be very specific to this board I'm afraid.

Kevin

**Reply by drhitchcockco** [reply]
Thanks for the swift reply. It sounds like it'll just be easier to set it up the recommended LCDC1602 Display and KY-040 rotary encoder.

Benet.
